optimizely/optimizely.py

```diff
@@ -174,8 +163,8 @@ def __init__(
             else:
                 self.config_manager = StaticConfigManager(**config_manager_options)
 
-        if not self.sdk_settings.odp_disabled:
-            self._update_odp_config_on_datafile_update()
+        self.odp_manager: OdpManager
+        self.setup_odp(sdk_key or self.config_manager.get_sdk_key())
```

Comment:
I see 2 changes related to "sdk_key" in here and in config_manager, both optional.
If none of them required, then we may have a case sdk_key is null.
As noted in my comment in config_manager, we may not need get_sdk_key in config_manager. 
We can consider "sdk_key" in optimizely from optional to required while removing the changes in config_manager.